projects/ad9656_fmc/src/app/ad9656_fmc.c

```diff
@@ -0,0 +1,265 @@
+		.user_test_pattern2 = 0xC3D4
+	};
+
+	ad9508_spi_param.platform_ops = &xil_platform_ops;
+	ad9508_spi_param.extra = &xil_spi_param;
+	ad9553_spi_param.platform_ops = &xil_platform_ops;
+	ad9553_spi_param.extra = &xil_spi_param;
+	ad9656_spi_param.platform_ops = &xil_platform_ops;
+	ad9656_spi_param.extra = &xil_spi_param;
+
+	struct ad9508_init_param	ad9508_param;
+	struct ad9553_init_param	ad9553_param;
+	struct ad9656_init_param	ad9656_param;
+
+	ad9508_param.spi_init = ad9508_spi_param;
+	ad9553_param.spi_init = ad9553_spi_param;
+	ad9656_param.spi_init = ad9656_spi_param;
+
+	struct ad9508_dev *ad9508_device;
+	struct ad9553_dev *ad9553_device;
+	struct ad9656_dev *ad9656_device;
+
+//******************************************************************************
+// setup the base addresses
+//******************************************************************************
+
+	struct adxcvr_init ad9656_xcvr_param = {
+		.name = "ad9656_xcvr",
+		.base = XPAR_AXI_AD9656_RX_XCVR_BASEADDR,
+		.sys_clk_sel = 0,
+		.out_clk_sel = 4,
+		.lpm_enable = 1,
+		.cpll_enable = 1,
+		.ref_rate_khz = 125000,
+		.lane_rate_khz = 2500000
+	};
+
+//******************************************************************************
+// ADC (AD9656) and the receive path ( AXI_ADXCVR,
+//	JESD204, AXI_AD9656, RX DMAC) configuration
+//******************************************************************************
+
+	/* JESD initialization */
+	struct jesd204_rx_init  ad9656_jesd_param = {
+		.name = "ad9656_jesd",
+		.base = RX_JESD_BASEADDR,
+		.octets_per_frame = 2,
+		.frames_per_multiframe = 32,
+		.subclass = 1,
+		.device_clk_khz = 2500000/40,
+		.lane_clk_khz = 2500000
+	};
+
+	struct axi_jesd204_rx *ad9656_jesd;
+
+	/* ADC Core */
+	struct axi_adc_init ad9656_core_param = {
+		.name = "ad9656_adc",
+		.base = RX_CORE_BASEADDR,
+		.num_channels = 4
+	};
+	struct axi_adc	*ad9656_core;
+	struct adxcvr	*ad9656_xcvr;
+
+//******************************************************************************
+// configure the receiver DMA
+//******************************************************************************
+
+	struct axi_dmac_init ad9656_dmac_param = {
+		.name = "ad9656_dmac",
+		.base = RX_DMA_BASEADDR,
+		.direction = DMA_DEV_TO_MEM,
+		.flags = 0
+	};
+	struct axi_dmac *ad9656_dmac;
+
+//******************************************************************************
+// bring up the system
+//******************************************************************************
+
+	ad9656_param.lane_rate_kbps = 2500000;
+
+	// setup clocks
+	if (ad9508_setup(&ad9508_device, &ad9508_param) != SUCCESS)
+		printf("The ad9508 chip could not be setup correctly!\n");
+	else
+		printf("The ad9508 chip successfully configured\n");
+
+	if (ad9553_setup(&ad9553_device, &ad9553_param) != SUCCESS)
+		printf("The ad9553 chip could not be setup!\n");
+	else
+		printf("The ad9553 chip successfully configured\n");
+
+	// ADC
+	ad9656_setup(&ad9656_device, &ad9656_param);
```

Comment:
I would check for this return value as well.

Reply:
Done in v6